### ISSUE TITLE
Lint/UselessMethodDefinition false positive with ruby 3

### ISSUE THREAD

**Issue by tycooon** [rubocop/rubocop#11415]
Let's say I have the following code:

```ruby
class A
  def self.foo(*args, **kwargs)
    [args, kwargs]
  end
end

class B < A
  def self.foo(*)
    super
  end
end

pp A.foo(smth: 42)
pp A.foo({ smth: 42 })
pp B.foo(smth: 42)
pp B.foo({ smth: 42 })
```

Rubocop reports the following warning:
```
/tmp/kwargs.rb:8:3: W: [Correctable] Lint/UselessMethodDefinition: Useless method definition detected.
  def self.foo(*) ...
  ^^^^^^^^^^^^^^^
```

This is true for Ruby 2. However, this code actually changes the behavior in Ruby 3 becase it converts kwargs into normal args in this case. Here are the actual results:

```
[[], {:smth=>42}]
[[{:smth=>42}], {}]
[[{:smth=>42}], {}]
[[{:smth=>42}], {}]
```

## RuboCop version

```
$ [bundle exec] rubocop -V
1.43.0 (using Parser 3.2.0.0, rubocop-ast 1.24.1, running on ruby 3.2.0) [arm64-darwin22]
```
